akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -50,20 +45,14 @@ class Encoder(
 
         send.senderOption match {
           case Some(sender) ⇒
-            //            val senderStr = senderCache.get(sender) match {
-            //              case null ⇒
-            //                val s = sender.path.toSerializationFormatWithAddress(localAddress)
-            //           //      FIXME we might need an efficient LRU cache, or replaced by compression table
-            //                if (senderCache.size() >= 1000)
-            //                  senderCache.clear()
-            //                senderCache.put(sender, s)
-            //                s
-            //              case s ⇒ s
-            //            }
-            headerBuilder.senderActorRef = sender // TODO cache not used now
+            // FIXME: Since we must include the loacl address, and API works on actor refs 
+            headerBuilder.senderActorRef = new MinimalActorRef {
```

Comment:
This was previously string based, thus it was `sender.path.toSerializationFormatWithAddress(localAddress)`.
Currently builder works on ActorRef, as it may store it in the table, so this workaround is in place...

I'll rethink when to require passing around ActorRef (into builder / tables) and when string repr is enough.